### PR TITLE
feat(text-field): reset value on type change

### LINKED ISSUE #6827: fix: Value of text-field is not updated when the text-field type changes
# 🐛 Bug Report

When changing the type of a `text-field` which already contains content, the rules of the new type are automatically applied by the browser. This causes an automatic value change of the inner input element. This change is not reflected back to users of the web component. This makes the value that is display differ from the value that is accessible by the developer. 

## 💻 Repro or Code Sample

I think this StackBlitz repro is the best explaination. 

Steps to reproduce:
1. Paste the following text in the text-field: "`        Demo Text with leading spaces.    ` ".
2. Open the JS console
3. Log the values using both buttons
4. Change the type of the text-field
    - Notice the visible change of the value
5. Log the values of the text-field again

## 🤔 Expected Behavior

The change of the inner input element should reflect back to the custom element and thus the user of the custom element.

## 😯 Current Behavior

The value of the inner input element is not reflected back to the custom element after an automatic value change due to the type change of the input.

## 💁 Possible Solution

Two-Way Binding of the value property.

## 🔦 Context

It doesn't affect me currently, as the type of an input field should not change after the user has set any value in my opinion. Even though this is an edge case, I think it is necessary to fix this behaviour.  
This behaviour could be really confusing, especially while developing automated UI tests.

## 🌍 Your Environment

* OS & Device: MacOS
* Browser Microsoft Edge
* Version "@microsoft/fast-components": "^2.30.6"

### PR DESCRIPTION
**Pull request by 43081j**
This changes `fast-text-field` to reset its value any time `type` changes.

The reason for this is that switching between two incompatible input types may reset the inner control's value, which won't otherwise be reflected to the outer element.

For example, using `type="text"` and typing a value into the input, then changing to `type="number"`. This should reset the value, since the inner control will have already done so.

### 🎫 Issues

Fixes #6827

## 👩‍💻 Reviewer Notes

In material, this is solved by resetting the value any time the component's properties change at all. Maybe we should do the same here? I just wasn't sure if there's a lifecycle callback in FAST for when _any_ properties change.

I'm also not too sure the test I've added is consistent with how you usually test this stuff. I didn't see any other tests testing the members of the element itself, only via attributes.

If you want to solve this another way, let me know and I'm happy to update.

## 📑 Test Plan

Added a new test for changing `type` after input has occurred. 

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: change/@microsoft-fast-foundation-fbee44f7-f673-4961-a65a-be11acf4b0eb.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
Comment (scomea) [inline change/@microsoft-fast-foundation-fbee44f7-f673-4961-a65a-be11acf4b0eb.json:2]: "prerelease"

File: change/@microsoft-fast-foundation-fbee44f7-f673-4961-a65a-be11acf4b0eb.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  "comment": "feat(text-field): reset value on type change",
+  "packageName": "@microsoft/fast-foundation",
+  "email": "43081j@users.noreply.github.com",
+  "dependentChangeType": "patch"
```
Comment (scomea) [inline change/@microsoft-fast-foundation-fbee44f7-f673-4961-a65a-be11acf4b0eb.json:6]: "prerelease"

**Review by scomea — APPROVED** [review]

**Review by scomea — CHANGES_REQUESTED** [review]
Seems reasonable, change file needs to be corrected tho.

**Comment by 43081j** [comment]
sorry only just got around to this

have rebased on master and updated the change file

**Comment by 43081j** [comment]
have rebased onto latest master

any chance anyone can take a look at merging this some day? 👀

**Review by bheston — APPROVED** [review]
